src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorProjectInfoSerializer.cs

```diff
@@ -204,4 +220,32 @@ private static bool TryGetFileKind(string? filePath, [NotNullWhen(true)] out str
         fileKind = null;
         return false;
     }
+
+    private static bool TryGetRazorFileName(string? filePath, [NotNullWhen(true)] out string? razorFilePath)
+    {
+        if (filePath is null)
+        {
+            razorFilePath = null;
+            return false;
+        }
+
+        // Must match C# extension: https://github.com/dotnet/vscode-csharp/blob/main/src/razor/src/razorConventions.ts#L10
+        const string prefix = "virtualcsharp-razor:///";
+        const string suffix = "__virtual.cs";
+
+        var path = filePath.AsSpan();
+
+        // Generated files have a path like: virtualcsharp-razor:///e:/Scratch/RazorInConsole/Goo.cshtml__virtual.cs
+        if (path.StartsWith(prefix, StringComparison.OrdinalIgnoreCase) &&
+            (path.EndsWith($".razor{suffix}", s_stringComparison) || path.EndsWith($".cshtml{suffix}", s_stringComparison)))
```

Comment:
Should `$".razor{suffix}"` and `$".cshtml{suffix}` be defined as consts above?

Reply:
I can do that. I only `const`'d the things that were used more than once, since the compiler inlines the const here anyway

Reply:
Aside: I wish it inlined the length of the const, below, too.